### PR TITLE
Select MPL instead of LGPL

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

File: tools/licenses/README.md [hunk 1]
```diff
@@ -20,9 +29,26 @@ rm -rf ../../../out/licenses
...
+the diffs to see what changed.
+
+```
+cp ../../../out/licenses/* ../../ci/licenses_golden
+git diff
+```
+
+If the only changes are to what files are included, then you're good
+to go. However, if any of the _licenses_ changed, whether new licenses
+are added, old ones removed, or any have their content changed in any
```
Comment (gspencergoog) [inline tools/licenses/README.md]: I would emphasize the "in any way" part.  E.g. people might assume whitespace changes don't constitute a "change".

File: tools/licenses/README.md [hunk 2]
```diff
@@ -20,9 +29,26 @@ rm -rf ../../../out/licenses
...
+way, you must update the actual license file.
+
+## IF ANYTHING SIGNIFICANT CHANGED, YOU MUST UPDATE THE ACTUAL LICENSE FILE AS WELL
```
Comment (gspencergoog) [inline tools/licenses/README.md]: What constitutes "significant"?  Best if that's not left open to interpretation.

File: tools/licenses/lib/main.dart [hunk 3]
```diff
@@ -2268,6 +2271,29 @@ Future<void> _collectLicensesForComponent(_RepositoryDirectory componentRoot, {
   sink.writeln('USED LICENSES:\n');
   final List<License> usedLicenses = licenses.where((License license) => license.isUsed).toList();
   final List<String> output = usedLicenses.map((License license) => license.toString()).toList();
+  for (int index = 0; index < output.length; index += 1) {
```
Comment (gspencergoog) [inline tools/licenses/lib/main.dart]: I saw the "Sanity Checks" section at the bottom that this replaces, but what's the rationale for each of these? Like, why are we excluding each of these?

Also, should they be more generalized to catch more cases (e.g. if someone wraps/indents the input text differently, are these going to miss things?)

File: tools/licenses/README.md (hunk 2 above)
Comment (Hixie) [reply tools/licenses/README.md]: it's defined in the paragraph above :-)

File: tools/licenses/lib/main.dart (hunk 3 above)
Comment (Hixie) [reply tools/licenses/lib/main.dart]: They're cases that I ran into when writing the script a few years ago, which are indicative of an unexpected change in the licensing situation. I'll add a comment.

File: tools/licenses/README.md (hunk 1 above)
Comment (Hixie) [reply tools/licenses/README.md]: updated README for this and other general improvements

File: tools/licenses/README.md (hunk 2 above)
Comment (Hixie) [reply tools/licenses/README.md]: i removed this header to make it clearer

Comment (gspencergoog) [reply tools/licenses/README.md]: OK, I guess, but it still feels like leaving it open for interpretation. Maybe this?:
```suggestion
## IF ANY LICENSES CHANGED, YOU MUST UPDATE THE ACTUAL LICENSE FILE AS WELL
```

**Comment by gspencergoog** [comment]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Pull request merged**